Tests/test_DNAconstruct.py

```diff
@@ -2,207 +2,85 @@
-        myconst2 = DNA_construct(newplist,circular=True)
-        myMixture2 = TxTlExtract(name = "txtl", parameters = parameters, \
-                            components = [myconst2])
-        myCRN2 = myMixture2.compile_crn()
-        #we add a promoter and more reactions and species are made
-        self.assertTrue(len(myCRN2.species)>numspec)
-        self.assertTrue(len(myCRN2.reactions)>numrxn)
-        #now there are two promoters
-        self.assertTrue(sum([spec.material_type=="rna" for spec in myCRN2.species])==2)
-
-        #make sure we saved the predicted values
-        mc2_from_mix = myMixture2.get_component(name=myconst2.name)
-        self.assertTrue(mc2_from_mix.predicted_rnas is not None)
-        self.assertTrue(mc2_from_mix.predicted_proteins is not None)
-        mc2_from_mix[4]=myt2.set_dir("forward")
-        #now, when we change the cosntruct, the previous values are not valid any more
-        self.assertTrue(mc2_from_mix.predicted_rnas is None)
-        self.assertTrue(mc2_from_mix.predicted_proteins is None)
-        
-
-
-
-
-
+def test_promoter_DNAconstruct():
+    P = Promoter("pconst") #constitutive promoter
+    parameters={"cooperativity":2,"kb":100, "ku":10, "ktx":.05, "ktl":.2, "kdeg":2,"kint":.05}
+    mechs = {"transcription":Transcription_MM(Species("RNAP",material_type="protein"))}
+
+    x = DNA_construct([P],mechanisms = mechs,parameters=parameters)
+    assert(not(P in x.parts_list)) #make sure P is copied
+    assert(P in x) #testing "contains" function of Construct
+    y = x.enumerate_components()
+    assert([x[0]] == y)
+    assert(x[0].transcript is None)
+
+    #works in reverse as well
+    x = DNA_construct([[P,"reverse"]],mechanisms = mechs,parameters=parameters)
+    assert(not(P in x.parts_list)) #make sure P is copied
+    assert(P in x) #testing "contains" function of Construct
+    y = x.enumerate_components()
+    assert([x[0]] == y)
+    assert(x[0].transcript is None)
+
+def test_promoter_terminator_DNAconstruct():
+    P = Promoter("pconst") #constitutive promoter
+    T = Terminator("term")
+    parameters={"cooperativity":2,"kb":100, "ku":10, "ktx":.05, "ktl":.2, "kdeg":2,"kint":.05}
+    mechs = {"transcription":Transcription_MM(Species("RNAP",material_type="protein"))}
+
+    #minimal RNA transcription
+    x = DNA_construct([P,T],mechanisms = mechs,parameters=parameters)
+    y = x.enumerate_components()
+    assert(y[0] == x[0]) #correct promoter is returned
+    assert(x[0].transcript == y[1].get_species()) #promoter in the DNA_construct has the correct transcript species
+    assert(y[1].promoter == x[0]) #rna is linked back to the DNA_construct's promoter
+    assert(y[1].promoter == y[0]) #correct promoter is returned by enumerate_constructs
+
+    #"incoherent" transcription
+    x = DNA_construct([[P,"reverse"],T],mechanisms = mechs,parameters=parameters)
+    y = x.enumerate_components()
```

Comment:
Can we have helpful comments in enumerate_components about what it returns? I think it is defined at multiple places but it wasn't really clear to me what this will return in different cases.

Reply:
(also because this function is the one that I see has been used so many times in tests)